crates/core/src/dht/dht_reducers.rs

```diff
@@ -138,14 +143,25 @@ pub(crate) fn reduce_hold_aspect(
                     reduce_remove_entry_inner(&mut new_store, &crud_link, &header.entry_address());
                 Some(new_store)
             } else {
-                error!("EntryAspect::Update without crud_link in header received!");
+                let err = "EntryAspect::Deletion without crud_link in header received!";
+                hold_result = Err(HolochainError::ErrorGeneric(err.to_string()));
                 None
             }
         }
         EntryAspect::Header(_) => {
-            error!("Got EntryAspect::Header which is not implemented.");
+            let err = "Got EntryAspect::Header which is not implemented.";
+            hold_result = Err(HolochainError::ErrorGeneric(err.to_string()));
             None
         }
+    };
+    if let Some(ref mut store) = r {
+        store.mark_aspect_as_held(&aspect);
+        store.mark_hold_aspect_complete(id.clone(), hold_result);
+        r
+    } else {
+        let mut store = (*old_store).clone();
+        store.mark_hold_aspect_complete(id.clone(), hold_result);
```

Comment:
Despite my last comment, I think it is a good thing to store the result of the holding workflow in the state as done here. (So we don't [always] rerun an already failed validation - and for debugging). I still think introducing the complexity of getting this result back to sim2h as a response for the holding request is too brittle and having sim2h (re-)poll for the gossip list (instead of marking something as hold because the request was sent) is the real fix here.

Reply:
No, that can't work in the end.  Remember that sim2h is where the intelligence lies of who should hold what.  It has the knowledge of who is in the sharding neighborhood.  The only reason it asks for a gossip list is to update it's understanding of who is actually holding what data so it can correctly forward queries.  But at the same time it should be making the decisions about who SHOULD hold what.  These fixes are about getting that right.

Reply:
How is this affecting the network code's reasoning about who should hold what?